pennylane_lightning/core/src/simulators/lightning_kokkos/catalyst/LightningKokkosSimulator.cpp

```diff
@@ -142,15 +142,17 @@ void LightningKokkosSimulator::SetState(DataView<std::complex<double>, 1> &data,
     std::size_t expected_wires = static_cast<std::size_t>(log2(data.size()));
     RT_ASSERT(expected_wires == wires.size());
     std::vector<Kokkos::complex<double>> data_vector(data.begin(), data.end());
-    std::vector<std::size_t> wires_size_t(wires.begin(), wires.end());
+    auto &&dev_wires = getDeviceWires(wires);
+    std::vector<std::size_t> wires_size_t(dev_wires.begin(), dev_wires.end());
 
     this->device_sv->setStateVector(data_vector, wires_size_t);
 }
 
 void LightningKokkosSimulator::SetBasisState(DataView<int8_t, 1> &data,
                                              std::vector<QubitIdType> &wires) {
     std::vector<std::size_t> basis_state(data.begin(), data.end());
-    std::vector<std::size_t> wires_size_t(wires.begin(), wires.end());
+    auto &&dev_wires = getDeviceWires(wires);
+    std::vector<std::size_t> wires_size_t(dev_wires.begin(), dev_wires.end());
```

Comment:
Do we need to make a copy from the provided r-value std::vector? Since they are both the same, can we just use the r-value to pass along?

Reply:
When we define it as `auto &&` is a `universal reference` and no `rvalue reference`, right? If the intention is to have values, the type needs to be specified.

Reply:
The updated name is forwarding [reference](https://en.cppreference.com/w/cpp/language/reference).

Reply:
how about just `this->device_sv->setBasisState(basis_state, getDeviceWires(wires))`? This way there's no extra in-between copy requiring references, and [setBasisState takes in const reference so no extra cost of copying](https://github.com/PennyLaneAI/pennylane-lightning/blob/4c6e4bb750b7d976fe661677895692ef6c129723/pennylane_lightning/core/src/simulators/lightning_kokkos/StateVectorKokkos.hpp#L153)

Reply:
Looks pretty good! Thanks!